app.js

```diff
@@ -425,15 +423,47 @@ io.use(function (socket, next) {
     });
 });
 
+io.use(function (socket, next) {
+    logger.info('openHAB-cloud: obtaining lock for connection for uuid ' + socket.handshake.uuid);
+    socket.connectionId = uuid.v1(); //we will check this when we handle disconnects
+    //set a lock so only one connection from the same client can connect, avoids split brain when clients reconnect
+    socket.redisLockKey = 'connection:' + socket.handshake.uuid;
+    redis.set(socket.redisLockKey, socket.connectionId, 'NX', 'EX', system.getConnectionLockTimeSeconds(), (err, result) => {
+        if(err) {
+            logger.info('openHAB-cloud: error attaining connection lock for  uuid ' + socket.handshake.uuid + ' connectionId ' + socket.connectionId + ' ' + err);
+            next(new Error('connection lock error'));
+        }
+
+        if(!result){
```

Comment:
Hmm, i assumed calling next(new Error('connection lock error'));`was enough (it will kill the connection) , but i'm guessing that eventually that will return and continue to process , so yes probably a good idea to guard against.